rosidl_typesupport_opendds_cpp/CMakeLists.txt

```
@@ -38,7 +38,7 @@ ament_python_install_package(${PROJECT_NAME})
 
 add_library(${PROJECT_NAME} SHARED)
 OPENDDS_TARGET_SOURCES(${PROJECT_NAME} src/identifier.cpp src/wstring_conversion.cpp)
```

```suggestion
add_library(${PROJECT_NAME} SHARED src/identifier.cpp src/wstring_conversion.cpp)
```

I would prefer vanilla CMake when ever possible. I don't really even like that this is possible. This also stops CMake from complaining about the library not having sources in this case.

---

With this approach, as we have found, OpenDDS settings (like the one for OpenDDS security) do not get forwarded to the library. So we decided to use OPENDDS_TARGET_SOURCES to add files wherever 'add_library' is used in this project.

---

I have already discussed this with Adam and this is a bug in the module. `OPENDDS_TARGET_SOURCES` should just process IDL files, nothing more.